### PR TITLE
godoc update for scheduler predicates.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:
This is a follow up PR for https://github.com/kubernetes/kubernetes/pull/46621 

/cc @timothysc 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -703,6 +707,7 @@ func PodMatchNodeSelector(pod *v1.Pod, meta interface{}, nodeInfo *schedulercach
 	return false, []algorithm.PredicateFailureReason{ErrNodeSelectorNotMatch}, nil
 }
 
+// PodMatchNodeSelector checks if a pod spec node name match the current node.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/PodMatchNodeSelector/PodFitsHost/

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -1233,6 +1244,7 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod, node
 	return true
 }
 
+// Determine if a pod tolertaions can tolerate the node taints
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: start with "PodToleratesNodeTaints".

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 3]
```diff
@@ -863,6 +868,7 @@ func (s *ServiceAffinity) checkServiceAffinity(pod *v1.Pod, meta interface{}, no
 	return false, []algorithm.PredicateFailureReason{ErrServiceAffinityViolated}, nil
 }
 
+// GeneralPredicates checks if a node has free ports for a specifed pod ports request.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/GeneralPredicates/PodFitsHostPorts

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 2 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 3 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 4]
```diff
@@ -565,6 +565,9 @@ func podName(pod *v1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
+// PodFitsResources checks if a node has sufficient resources, such as cpu, memory, gpu, opaque int resources etc to run a pod.
+// First return value indicates whether a node has sufficient resources to run a pod while the second return value indicates the
+// predicate failure reasons if the node does have insufficient resources to run the pod.
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: nit: s/does have/has

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 5]
```diff
@@ -692,6 +695,7 @@ func podMatchesNodeLabels(pod *v1.Pod, node *v1.Node) bool {
 	return nodeAffinityMatches
 }
 
+// PodMatchNodeSelector checks if a pod node selector match the node label.
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/match/matches

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 6]
```diff
@@ -703,6 +707,7 @@ func PodMatchNodeSelector(pod *v1.Pod, meta interface{}, nodeInfo *schedulercach
 	return false, []algorithm.PredicateFailureReason{ErrNodeSelectorNotMatch}, nil
 }
 
+// PodFitsHost checks if a pod spec node name match the current node.
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/match/matches

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 7]
```diff
@@ -863,6 +868,7 @@ func (s *ServiceAffinity) checkServiceAffinity(pod *v1.Pod, meta interface{}, no
 	return false, []algorithm.PredicateFailureReason{ErrServiceAffinityViolated}, nil
 }
 
+// PodFitsHostPorts checks if a node has free ports for a specifed pod ports request.
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/a specified pod ports request/the requested pod ports

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 8]
```diff
@@ -896,6 +902,8 @@ func haveSame(a1, a2 []string) bool {
 	return false
 }
 
+// GeneralPredicates includes noncriticalPredicates and EssentialPredicates. noncriticalPredicates are the predicates that
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/GeneralPredicates includes noncriticalPredicates and EssentialPredicates./GeneralPredicates checks whether noncriticalPredicates and EssentialPredicates pass.

**Review by bsalamat — COMMENTED** [review]
Thanks, @gyliu513 for the improvement. I have a few minor comments.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 4 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 5 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 6 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 7 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 8 above)
Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Done

**Review by gyliu513 — COMMENTED** [review]
Thanks @bsalamat

**Comment by gyliu513** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by gyliu513** [comment]
How about make this as milestone 1.7 so that it can be in submit queue?  @bsalamat @k82cn

**Comment by k82cn** [comment]
The 1.7 is freeze, this PR is OK to be 1.8; it's not a functional bug :). Non-1.7 PR will be merged after 1.7 release, the queue will accept it.

**Comment by gyliu513** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by gyliu513** [comment]
ping @bsalamat @jayunit100 for a /lgtm , thanks.

**Pull request merged**